babynames.py

```diff
@@ -42,29 +45,66 @@
 def extract_names(filename):
     """
     Given a single file name for babyXXXX.html, returns a single list starting
-    with the year string followed by the name-rank strings in alphabetical order.
+    with the year string followed by the name-rank strings in alphabetical
+    order.
     ['2006', 'Aaliyah 91', Aaron 57', 'Abagail 895', ' ...]
     """
     names = []
-    # +++your code here+++
-    return names
+
+    names.append(re.findall("\d\d\d\d", filename)[0])
+
+    with open(filename, "r") as f:
+        f = f.read()
+
+    names_list = re.findall("<td>(.+?)</td>", f)
+
+    names_dict = {}
+
+    for index in range(len(names_list)):
+        if index % 3 != 0:
```

Comment:
Clever! I didn't think of taking advantage of the multiples of 3.  One thing you could have done here is run the list backward so that any over-writing that happened would be of the smaller number over the larger number.  Just an idea.  :nerd_face: